src/main/kotlin/com/ekino/oss/recaptcha/config/ReCaptchaFilter.kt

```diff
@@ -0,0 +1,49 @@
+package com.ekino.oss.recaptcha.config
+
+import com.ekino.oss.recaptcha.exception.MissingResponseException
+import com.ekino.oss.recaptcha.exception.ReCaptchaValidationException
+import com.ekino.oss.recaptcha.service.ReCaptcaValidationResult
+import com.ekino.oss.recaptcha.service.ReCaptchaValidationService
+import org.springframework.web.filter.OncePerRequestFilter
+import javax.servlet.FilterChain
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+internal const val BY_PASS_VALIDATION_HEADER_NAME = "X-ReCaptcha-ByPass-Key"
+
+/**
+ * Custom filter for validating reCaptcha.
+ * Will retrieve reCatpcha response from header or request parameter from {@code responseName}.
+ * Depending on configuration, validation will be applied only for specific method.
+ * Validation can be by-passed passing a custom header with a specific key previously configured.
+ */
+class ReCaptchaFilter(
+  private val validationService: ReCaptchaValidationService,
+  private val responseName: String,
+  private val byPassKey: String?,
+  private val filteredMethods: Set<String>
+) : OncePerRequestFilter() {
+
+  override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
+    if (!byPassReCaptchaValidation(request) && requestMethodIsFiltered(request)) {
+      logger.debug { "[ReCaptcha] ${request.requestURI} is protected by reCaptcha. ReCaptcha response will be validated." }
+
+      val reCaptchaResponse = request.getParameter(responseName)
```

Comment:
Maybe it could be good to add a paragraph in README.md to explain where the response is retrieved (query param  first then header) and the name of the looked parameter "g-recaptcha-response" (that could be overridden by configuration).

Reply:
Yes the documentation still need some modifications I will update it

Reply:
Done